CoreService/Workflow.psm1

```diff
@@ -0,0 +1,231 @@
+		[Parameter(ParameterSetName='Status')]
+		[ValidateSet('', 'Active', 'Historical')]
+		[string]$Status,
+		
+		# The TCM URI of the Publication Target to load.
+        [Parameter(ParameterSetName='AssignedToById')]
+		[ValidateNotNullOrEmpty()]
+        [string]$AssignedToById,
+		
+		# The Title of the Publication Target to load. This is slower than specifying the ID.
+        [Parameter(ParameterSetName='AssignedToByTitle')]
+		[ValidateNotNullOrEmpty()]
+        [string]$AssignedToByTitle
+	)		
+	
+	Begin
+	{
+        $client = Get-TridionCoreServiceClient -Verbose:($PSBoundParameters['Verbose'] -eq $true);
+	}
+	
+    Process
+    {
+		$filter = New-Object Tridion.ContentManager.CoreService.Client.ProcessesFilterData;
+		$acvities = new-object Tridion.ContentManager.CoreService.Client.ActivityData;
+		$readOption = New-Object Tridion.ContentManager.CoreService.Client.ReadOptions;
+		$readOption.LoadFlags = [Tridion.ContentManager.CoreService.Client.LoadFlags]::None;
+		$processInstanceData = New-Object Tridion.ContentManager.CoreService.Client.ProcessInstanceData;
+		$list = New-Object -TypeName System.Collections.Generic.List[Tridion.ContentManager.CoreService.Client.ProcessInstanceData];
+					
+        if ($client -ne $null)
+        {
+			switch($PsCmdlet.ParameterSetName)
+			{
+			    'Status'
+				{
+					Write-Verbose "Loading list of Workflow Items...";
+					
+					if($Status -eq $null)
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Any;				
+					}
+					if($Status -eq "Active")
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;				
+					}
+					if($Status -eq "Historical")
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Historical;				
+					}			
+								
+					$WorkflowItemDetail = $client.GetSystemWideList($filter);					
+					return $WorkflowItemDetail
+				}
+			    'AssignedToById'
+				{
+					Write-Verbose "Loading Publication Target with Id '$AssignedToById'..."
+					if ($AssignedToById -ne $null)
+					{
+						if(!$AssignedToById.EndsWith('-65552'))
+						{						
+							if(!$AssignedToById.EndsWith('-65568'))
+							{
+								Write-Error "'$AssignedToById' is not a valid User or Group TCM URI.";
+								return;
+							}
+							else
+							{
+								$groupData = Get-TridionGroup -Id $AssignedToById
+							    if($groupData -eq $null)
+								{
+									Write-Error "'$AssignedToById' is not a valid Group TCM URI.";
+									return;
+								}
+							}
+						}
+						else
+						{
+							$userData = Get-TridionUser -Id $AssignedToById
+							if($userData -eq $null)
+							{
+								Write-Error "'$AssignedToById' is not a valid User TCM URI.";
+								return;
+							}							
+						}					
+					}
+					$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;
+					$WorkflowItemDetail = $client.GetSystemWideList($filter);
```

Comment:
In general, we shouldn't use the client directly -- as it cannot be tested with Pester. Use the utility functions instead (which can be mocked up and tested) - such as _GetSystemWideList (in Utilities.ps1)